extension/encoding/googlecloudlogentryencodingextension/internal/armorlog/parser.go

```diff
@@ -0,0 +1,335 @@
+	gcpArmorSecurityPolicyPriority = "gcp.armor.security_policy.priority"
+	// gcpArmorSecurityPolicyConfiguredAction holds  the name of the configured action in the matching rule
+	gcpArmorSecurityPolicyConfiguredAction = "gcp.armor.security_policy.configured_action"
+	// gcpArmorSecurityPolicyOutcome holds the outcome of executing the configured action
+	gcpArmorSecurityPolicyOutcome = "gcp.armor.security_policy.outcome"
+
+	// Rate limit action attributes
+	// gcpArmorRateLimitActionKey holds the the rate limit key value (up to 36 bytes)
+	gcpArmorRateLimitActionKey = "gcp.armor.security_policy.rate_limit.action.key"
+	// gcpArmorRateLimitActionOutcome holds the outcome of the rate limit action
+	gcpArmorRateLimitActionOutcome = "gcp.armor.security_policy.rate_limit.action.outcome"
+
+	// Extended attributes
+	// gcpArmorPreconfiguredExprIds holds the IDs of all preconfigured WAF rule expressions that triggered the rule
+	gcpArmorPreconfiguredExprIds = "gcp.armor.security_policy.preconfigured.expr_ids"
+	// gcpArmorThreatIntelligenceCategories holds information about the matched IP address lists from Google Threat Intelligence
+	gcpArmorThreatIntelligenceCategories = "gcp.armor.security_policy.threat_intelligence.categories"
+	// gcpArmorAddressGroupNames holds the names of the matched address groups
+	gcpArmorAddressGroupNames = "gcp.armor.security_policy.address_group.names"
+
+	// Enforced policy attributes
+	// gcpArmorAdaptiveProtectionAutoDeployAlertID holds the alert ID of the events that Adaptive Protection detected
+	gcpArmorAdaptiveProtectionAutoDeployAlertID = "gcp.armor.security_policy.adaptive_protection.auto_deploy.alert_id"
+
+	// Security policy request data attributes
+	// gcpArmorRecaptchaActionTokenScore holds the legitimacy score embedded in a of the reCAPTCHA action-token
+	gcpArmorRecaptchaActionTokenScore = "gcp.armor.request_data.recaptcha_action_token.score"
+	// gcpArmorRecaptchaSessionTokenScore holds the legitimacy score embedded in a of the reCAPTCHA session-token
+	gcpArmorRecaptchaSessionTokenScore = "gcp.armor.request_data.recaptcha_session_token.score"
+	// gcpArmorUserIPInfoSource holds a field that is typically the header from which the user IP was resolved
+	gcpArmorUserIPInfoSource = "http.request.header.source"
+	// gcpArmorRemoteIPInfoRegionCode holds the two-letter country code or region code for the IP address
+	gcpArmorRemoteIPInfoRegionCode = "geo.region.iso_code"
+	// gcpArmorRemoteIPInfoAsn holds the five-digit autonomous system number (ASN) for the IP address
+	gcpArmorRemoteIPInfoAsn = "gcp.armor.request_data.remote_ip.asn"
+	// gcpArmorTLSJa4Fingerprint holds a JA4 TTL/SSL fingerprint if the client connects using HTTPS, HTTP/2, or HTTP/3
+	gcpArmorTLSJa4Fingerprint = "tls.client.ja4"
+	// gcpArmorTLSJa3Fingerprint holds a JA3 TTL/SSL fingerprint if the client connects using HTTPS, HTTP/2, or HTTP/3
+	gcpArmorTLSJa3Fingerprint = "tls.client.ja3"
```

Comment:
Import from semantic conventions, like this:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e0f0c39a34eb753c1c04b36165507ac026766945/extension/encoding/googlecloudlogentryencodingextension/internal/auditlog/parser.go#L251

And then same comment for every field that is present in the conventions